src/Markdown/Editor/Impl/Utility/MarkdownUtility.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Text;
+using Microsoft.Common.Core;
+using Microsoft.Languages.Core.Text;
+
+namespace Microsoft.Markdown.Editor.Utility {
+    public static class MarkdownUtility {
+        /// <summary>
+        /// Converts R code block as it appears in R Markdown to legal R.
+        /// Drops options setting block '{r, x = FALSE, ...}.
+        /// Allows for curly brace nesting.
+        /// </summary>
+        /// <param name="content"></param>
+        /// <remarks>
+        /// http://rmarkdown.rstudio.com/developer_parameterized_reports.html#accessing_from_r
+        /// </remarks>
+        public static string GetRContentFromMarkdownCodeBlock(string content) {
+            while (true) {
+                // Locate start of the block
+                var start = content.IndexOfIgnoreCase("{r");
```

Comment:
Yes, but this should never execute more than once. that while() is not needed. Removed and added couple more tests